homeassistant/components/tuya/climate.py

```diff
@@ -436,8 +436,16 @@ def hvac_mode(self) -> str:
                 return self.entity_description.switch_only_hvac_mode
             return HVAC_MODE_OFF
 
-        if self.device.status.get(DPCode.MODE) is not None:
-            return TUYA_HVAC_TO_HA[self.device.status[DPCode.MODE]]
+        if (mode := self.device.status.get(DPCode.MODE)) is not None:
+            if mode not in TUYA_HVAC_TO_HA:
+                LOGGER.warning(
```

Comment:
We don't want side effects like logging in entity state properties. Can we add an update callback that handles it instead?

Reply:
Oh right, this will spam the logs 🤕 

I'll figure something out.

Reply:
Adjusted it; it now logs all unknown HVAC modes on set up (once).
Adjusted the logic for that.